ui/apps/platform/src/Containers/Vulnerabilities/PlatformCves/Overview/ClustersTableContainer.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import { Bullseye, Divider, Spinner } from '@patternfly/react-core';
+
+import TableErrorComponent from 'Components/PatternFly/TableErrorComponent';
+import useURLPagination from 'hooks/useURLPagination';
+
+import TableEntityToolbar, { TableEntityToolbarProps } from '../../components/TableEntityToolbar';
+import { QuerySearchFilter } from '../../types';
+
+export type ClustersTableContainerProps = {
+    filterToolbar: TableEntityToolbarProps['filterToolbar'];
+    entityToggleGroup: TableEntityToolbarProps['entityToggleGroup'];
+    querySearchFilter: QuerySearchFilter;
+    isFiltered: boolean;
+    rowCount: number;
+    pagination: ReturnType<typeof useURLPagination>;
+};
+
+function ClustersTableContainer({
+    filterToolbar,
+    entityToggleGroup,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    querySearchFilter,
+    isFiltered,
+    rowCount,
+    pagination,
+}: ClustersTableContainerProps) {
+    // TODO - Placeholders for query results
+    const data = [];
+    const previousData = undefined;
+    const error: Error | undefined = undefined;
+    const loading = false;
+
+    const tableData = data ?? previousData;
+
+    return (
+        <>
+            <TableEntityToolbar
+                filterToolbar={filterToolbar}
+                entityToggleGroup={entityToggleGroup}
+                pagination={pagination}
+                tableRowCount={rowCount}
+                isFiltered={isFiltered}
+            />
+            <Divider component="div" />
+            {loading && !tableData && (
+                <Bullseye>
+                    <Spinner isSVG />
+                </Bullseye>
+            )}
+            {error && (
```

Comment:
Help my turtle brain here, especially because this is a reference rendering pattern worthy to follow.
1. If initial request has an error, render the error after no longer loading, as expected.
2. If interaction in toolbar causes subsequent request, render loading spinner **and** previous error until either failure or success is known?

Reply:
For point 2 - Apollo's `useQuery` (and our internal `useRestQuery` used elsewhere) both immediately clear the error message when a new request is in flight. So there will never be a state where `loading === true` and `error !== undefined`. In this case, the subsequent request would immediately render the loading spinning until another failure or success occurs.

Reply:
Aha, super!